src/diffusers/models/attention.py

```diff
@@ -137,6 +137,7 @@ def __init__(self, dim, n_heads, d_head, dropout=0.0, context_dim=None, gated_ff
         self.checkpoint = checkpoint
 
     def forward(self, x, context=None):
+        x = x.contiguous() if x.device.type == "mps" else x
```

Comment:
can't we always do `x.contiguous()` on other devices as well ?

Reply:
We can, but my approach was to be as transparent as possible with the way it works in other devices. In this case I also wanted to prevent a memory copy. In addition, we might want to explore "channels last" format in the future (not sure if it would help here), and I wasn't sure about the implications of using contiguous if we do.

But happy to change it if it's clearer that way.

Reply:
`x.contiguous()` allocates new memory, so happy trying to avoid it if possible :-)